core/pom.xml

```diff
@@ -119,6 +119,70 @@
     </dependency>
   </dependencies>
 
+  <profiles>
+   <profile>
+      <id>swift</id>
+      <activation>
+        <property>
+          <name>swift.profile</name>
+          <value>swift</value>
+        </property>
+      </activation>
+      <properties>
+        <ufs></ufs>
+      </properties>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-openstack</artifactId>
+           <version>${hadoop-openstack.version}</version>
+        </dependency>
+      </dependencies>
+    </profile>
+    <profile>
+      <id>default</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>3.2</version>
+            <executions>
+              <execution>
+                <id>default-testCompile</id>
+                <phase>test-compile</phase>
+                <configuration>
+                  <testExcludes>
+                    <exclude>**/hadoop/contract/**</exclude>
+                  </testExcludes>
+                  <compilerArgs>
+                    <arg>-Xlint:-unchecked</arg>
+                  </compilerArgs>
+                </configuration>
+                <goals>
+                  <goal>testCompile</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
```

Comment:
You should be able to add a simpler profile to `underfs/hdfs/pom.xml` just including the open-stack dependency.